app/views/_macros/components.nunjucks

```diff
@@ -1,28 +1,114 @@
-      />
-    <!--<![endif]-->
-    <!--[if gt IE 7]><!--><noscript><!--<![endif]-->
-      {% set fallback = options.srcset[0] | split(' ') %}
-      <img src="{{ asset_path(fallback[0]) }}" alt="{{ options.alt }}" />
-    <!--[if gt IE 7]><!--></noscript><!--<![endif]-->
-  {% endif %}
+{% macro callout(props) %}
+  <section class="callout callout--{{ props.variant }}">
+    {% for component in props.children %}
+      {{ loadComponent(component.type)(component.props) }}
+    {% endfor %}
+  </section>
 {% endmacro %}
 
-{% macro markdown(content) %}
+{% macro reveal(props) %}
+  <details class="{{ 'details--inline' if props.variant == 'inline' }}">
+    <summary data-analytics="summary">
+      <span class="details__summary">
+        {% if props.variant == 'inline' %}
+          {{ props.summary.replace('[', '<span class="details__cta">').replace(']', '</span>') | safe }}
+        {% else %}
+          {{ props.summary }}
+        {% endif %}
+      </span>
+    </summary>
+
+    <div>
+      {% for component in props.children %}
+        {{ loadComponent(component.type)(component.props) }}
+      {% endfor %}
+    </div>
+  </details>
+{% endmacro %}
+
+{% macro text(props) %}
   <div class="reading-width">
-    {% markdown %}
-      {{ content | renderString | safe }}
-    {% endmarkdown %}
+    {% if props.variant == 'markdown' %}
+      {% markdown %}
+        {{ props.value | renderString | safe }}
+      {% endmarkdown %}
+    {% elseif props.variant == 'html' %}
+      {{ props.value | safe}}
+    {% elseif props.variant == 'plain' %}
+      {{ props.value }}
+    {% endif %}
+  </div>
+{% endmacro %}
+
+{% macro split_content(props) %}
+  <div class="panel">
+    <div class="panel__content panel__content--half">
+      {% for component in props.children[0] %}
+        {{ loadComponent(component.type)(component.props) }}
+      {% endfor %}
+    </div>
+
+    <div class="panel__content panel__content--half">
+      {% for component in props.children[1] %}
+        {{ loadComponent(component.type)(component.props) }}
+      {% endfor %}
+    </div>
   </div>
 {% endmacro %}
 
+{% macro panel(props) %}
+  <article class="panel{{ ' reading-width' if props.reading_width }}">
+    {% if props.header %}
+      <div class="panel__header">
+        {% for component in props.header %}
+          {{ loadComponent(component.type)(component.props) }}
+        {% endfor %}
+      </div>
+    {% endif %}
+
+    <div class="panel__content">
+      {% for component in props.body %}
+        {{ loadComponent(component.type)(component.props) }}
+      {% endfor %}
+    </div>
+
+    {% if props.footer %}
+      <footer class="panel__footer">
+        {% for component in props.footer %}
+          {{ loadComponent(component.type)(component.props) }}
+        {% endfor %}
+      </footer>
+    {% endif %}
+  </article>
+{% endmacro %}
+
+{% macro image(props) %}
+  {% if props.srcset %}
+    <figure class="card">
```

Comment:
Maybe we rename the `card` component to `figure` as that's the only place it's being used. Or something more generic like `media`?